certification/internal/shell/imagesource_registry.go

```diff
@@ -0,0 +1,79 @@
+package shell
+
+import (
+	"bytes"
+	"strings"
+
+	cmdchain "github.com/rainu/go-command-chain"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	log "github.com/sirupsen/logrus"
+)
+
+type ImageSourceRegistryCheck struct {
+}
+
+var approvedRegistries = map[string]string{
+	"registry.connect.dev.redhat.com":   "registry.connect.dev.redhat.com",
+	"registry.connect.qa.redhat.com":    "registry.connect.qa.redhat.com",
+	"registry.connect.stage.redhat.com": "registry.connect.stage.redhat.com",
+	"registry.connect.redhat.com":       "registry.connect.redhat.com",
+	"registry.redhat.io":                "registry.redhat.io",
+	"registry.access.redhat.com":        "registry.access.redhat.com",
+}
+
+func (p *ImageSourceRegistryCheck) Validate(bundleImage string) (bool, error) {
+
+	output := &bytes.Buffer{}
+	inputContent := strings.NewReader(bundleImage)
```

Comment:
if you are not using `inputContent`, please remove it and invoke the `strings.NewReader` directly on line 30.

Reply:
Nice